### PR TITLE
chore: Merge with latest master changes updating SDK to v20240530002

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-260124

**Note**: for reviewing suggest to look into the commits done after the merge commit. Also left comments on relevant changes.

Test group failing `cluster_outage_simulation` is currently also failing in master nightly runs.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,7 +25,7 @@ require (
 	github.com/zclconf/go-cty v1.14.4
 	go.mongodb.org/atlas v0.36.0
 	go.mongodb.org/atlas-sdk/v20231115014 v20231115014.0.0
-	go.mongodb.org/atlas-sdk/v20240530001 v20240530001.0.1-0.20240628150242-4d7b444ed046
+	go.mongodb.org/atlas-sdk/v20240530002 v20240530002.0.1-0.20240703083409-9e77f6cad45f
```
Comment (AgustinBettati) [inline go.mod:28]: - `v20231115014` is preserved to use old cluster endpoints
- `v20240530002` is adjusted to use lastest preview which includes new cluster API

File: internal/service/cloudbackupsnapshot/resource_cloud_backup_snapshot_test.go [hunk 2]
```diff
@@ -67,6 +67,7 @@ func TestAccBackupRSCloudBackupSnapshot_basic(t *testing.T) {
 }
 
 func TestAccBackupRSCloudBackupSnapshot_sharded(t *testing.T) {
+	acc.SkipTestForCI(t) // TODO: CLOUDP-260154 for ensuring this use case is supported
 	var (
 		projectID       = acc.ProjectIDExecution(t)
 		clusterName     = acc.RandomClusterName()
```
Comment (AgustinBettati) [inline internal/service/cloudbackupsnapshot/resource_cloud_backup_snapshot_test.go:73]: When running all tests as part of this PR detected a failing test in `cloud_backup_snapshot`. This is related to an advanced_cluster config using the old schema structure for defining a symmetric sharded cluster. This has to be supported so added an additional test in advanced_cluster, skipped both, and following up with a ticket until this case get resolved (detected incorrect behaviour in the API).

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 3]
```diff
@@ -487,6 +488,31 @@ func TestAccClusterAdvancedClusterConfig_selfManagedShardingIncorrectType(t *tes
 	})
 }
 
+func TestAccClusterAdvancedClusterConfig_symmetricGeoShardedOldSchema(t *testing.T) {
+	acc.SkipTestForCI(t) // TODO: CLOUDP-260154 for ensuring this use case is supported
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName = acc.RandomProjectName() // No ProjectIDExecution to avoid cross-region limits because multi-region
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:495]: new test that was added

**Review by lantoli — APPROVED** [review]
great job!

**Comment by lantoli** [comment]
cluster_outage_simulation is also failing in master, we'll need to fix and maybe move to this branch.  (you can go ahead and merge if only these tests fail)

**Review by marcosuma — APPROVED** [review]

**Pull request merged**